tsconfig.json

```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    // Target es5 for syntax (IE 11), but expect Vue projects (i.e. Babel) to provide polyfills up to at least es2016
+		// See https://stackoverflow.com/a/57474312/4167438 for explanation
+		"target": "es5",
+		"lib": [
+			"ES2016",
+			"DOM"
+		],
+    "module": "ESNext",
+    "strict": true,
```

Comment:
Please don't do this. Libraries are beholden to the author–not the other way around.

Reply:
Oh man my styling on this file looks super broken. Not sure how that happened 🤔 

And sounds good - for the most part, I was just copying over other existing `tsconfig` files we've used. Is it primarily the `noImplicitAny` that you're concerned about? Happy to turn that off. And also `strict` if we'd like to skip the null checking involved as well :+1:

Reply:
I find it a more efficient and less hassle approach to generally go with the least restrictive approach most of the time and only increase restriction once not doing so is causing more problems than doing so. This results in smaller, cleaner, more streamlined code that also better-reaches a broader audience. For maximum benefit at the least cost, I would simply "type" what would be unintuitive and/or complex for the consumers' benefit and then anything that is truly a gotcha for developers of this library. If the number of devs and the size of this codebase starts getting to the point where locking things down actually helps more than it costs, then I'd take a look at doing that when the problem actually presents itself.